pyiron_base/generic/inputlist.py

```diff
@@ -206,6 +206,8 @@ def __setitem__(self, key, val):
         key = _normalize(key)
 
         if isinstance(key, tuple):
+            if key[0] not in self.keys():
+                self[key[0]] = InputList()
```

Comment:
```suggestion
                self[key[0]] = type(self)()
```
This plays nicer with sub-classes.